src/pybamm/expression_tree/operations/serialise.py

```diff
@@ -399,3 +604,315 @@ def _convert_options(self, d):
+            )
+            json_dict = {
+                "type": "IndefiniteIntegral",
+                "children": [Serialise.convert_symbol_to_json(symbol.child)],
+                "integration_variable": Serialise.convert_symbol_to_json(
+                    integration_var
+                ),
+            }
+        elif isinstance(symbol, pybamm.BoundaryValue):
+            json_dict = {
+                "type": "BoundaryValue",
+                "side": symbol.side,
+                "children": [Serialise.convert_symbol_to_json(symbol.orphans[0])],
+            }
+        elif isinstance(symbol, pybamm.SpecificFunction):
+            if symbol.__class__ == pybamm.SpecificFunction:
+                raise NotImplementedError("SpecificFunction is not supported directly")
+            json_dict = {
+                "type": symbol.__class__.__name__,
+                "children": [
+                    Serialise.convert_symbol_to_json(c) for c in symbol.children
+                ],
+            }
+
+        elif isinstance(symbol, pybamm.UnaryOperator | pybamm.BinaryOperator):
+            json_dict = {
+                "type": symbol.__class__.__name__,
+                "children": [
+                    Serialise.convert_symbol_to_json(c) for c in symbol.children
+                ],
+            }
+
+        elif isinstance(symbol, pybamm.Symbol):
+            # Generic fallback for other symbols with children
+            json_dict = {
+                "type": symbol.__class__.__name__,
+                "domains": symbol.domains,
+                "children": [
+                    Serialise.convert_symbol_to_json(c) for c in symbol.children
+                ],
+            }
+            if hasattr(symbol, "name"):
+                json_dict["name"] = symbol.name
+
+        else:
+            raise ValueError(
+                f"Error processing '{symbol.name}'. Unknown symbol type: {type(symbol)}"
+            )
+        return json_dict
+
+    @staticmethod
+    def convert_symbol_from_json(json_data):
+        """
+        Recursively reconstructs a PyBaMM symbolic expression from a JSON dictionary.
+
+        Supports all major PyBaMM symbol types, including Scalars, Variables, Parameters,
+        Operators, FunctionParameters, Broadcasts, Interpolants.
+
+        Parameters
+        ----------
+        json_data : dict
+            A JSON-serialized representation of a PyBaMM expression, produced
+            by `Serialise.convert_symbol_to_json`.
+
+        Returns
+        -------
+        pybamm.Symbol or primitive
+            The reconstructed PyBaMM symbolic expression or a primitive (float, int, bool).
+
+        Examples
+        --------
+        >>> import pybamm
+        >>> from pybamm.expression_tree.operations.serialise import Serialise
+        >>> json_expr = {'type': 'Scalar', 'value': 42}
+        >>> Serialise.convert_symbol_from_json(json_expr)
+        pybamm.Scalar(42)
```

Comment:
Should probably ignore this as a doctest because of the memory address
```suggestion
        Scalar(0x1aa1e910749e9f89, 42.0, children=[], domains={})
```